gemfirexd/client/src/main/java/com/pivotal/gemfirexd/internal/client/net/NetConnection.java

```diff
@@ -2360,6 +2360,7 @@ else if (agent != null) {
               final String password = getDeferredResetPassword();
               flowConnect(password, this.securityMechanism_, false);
 
+              setCurrentSchema();
```

Comment:
Yes this is required. Just below notice the completeReset() call. As part of complete reset it tries to prepares the statements in the cache and their only the schema trouble starts happening. So in this reconnect case having only below will be too late. May be what I can do is that if it is called once I can skip the later one.